src/users/users.controller.spec.ts

```diff
@@ -0,0 +1,126 @@
+import { NotFoundException, HttpException, HttpStatus } from '@nestjs/common';
+import { Test, TestingModule } from '@nestjs/testing';
+import { RegisterDTO } from '../auth/auth.dto';
+import { UsersController } from './users.controller';
+import { UsersService } from './users.service';
+
+describe('UsersController', () => {
```

Comment:
testy na kontrolery widziałbym bardziej w formie e2e (używając biblioteki supertest). To pozwoliłoby sprawdzić, czy wysyłane żądanie jest poprawnie parsowane, czy odpowiedź zawiera odpowiedni JSON itd. Chodzi mi o coś takiego:
https://docs.nestjs.com/fundamentals/testing#end-to-end-testing
Można oczywiście w takim teście zamockować serwis żeby zwracał odpowiednie wartości tak jak już zrobiłeś